Add new package "qtcreatorcdbext" for Qt Creator's CDB Extension

michaelweghorn opened the pull request:
This adds the CDB debugger extension for Qt Creator. As discussed in #216, a new package is created rather than addint it into the existing `qtcreator` package.

The packaging files were created based on the ones for the 'qtcreator' package, i.e. those for the existing 'qtcreator' package  were initially copied, then adapted for this new package as needed.

Fixes: #216

Note: I've created this for Qt Creator 4.13.2 already, which is currently only available on the `master` branch, not `develop` yet, so this should probably not be integrated into the `develop` branch before PR #273.

AdmiringWorm commented on automatic/qtcreatorcdbext/Readme.md:
```diff
@@ -0,0 +1,3 @@
+# [<img src="https://cdn.jsdelivr.net/gh/AdmiringWorm/chocolatey-packages@35eb3e49edb3411d779f64e57ef023abac8a3a06/icons/qtcreator.png" height="48" width="48" /> ![Qt Creator CDB Extension](https://img.shields.io/chocolatey/v/qtcreator.svg?label=Qt%20Creator%20CDB%20Extension&style=for-the-badge)](https://chocolatey.org/packages/qtcreatorcdbext)
```
It looks like you are using targeting the the qtcreator package for this badge.

Please change `qtcreator.svg` to `qtcreator-cdbext.svg`

michaelweghorn replied:
Done

AdmiringWorm commented on automatic/qtcreatorcdbext/Readme.md:
```diff
@@ -0,0 +1,3 @@
+# [<img src="https://cdn.jsdelivr.net/gh/AdmiringWorm/chocolatey-packages@35eb3e49edb3411d779f64e57ef023abac8a3a06/icons/qtcreator.png" height="48" width="48" /> ![Qt Creator CDB Extension](https://img.shields.io/chocolatey/v/qtcreator.svg?label=Qt%20Creator%20CDB%20Extension&style=for-the-badge)](https://chocolatey.org/packages/qtcreatorcdbext)
+
+Qt Creator CDB Extension adds support for the Microsoft Console Debugger (CDB) to Qt Creator, a cross-platform, complete integrated development environment (IDE) for application developers to create applications for multiple desktop and mobile device platforms, such as Android and iOS.
```
I think perhaps we may need a note here in the description, from what I gather there is a need for Visual Studio 2012+ due to a need to have `cdb.exe` available.

michaelweghorn replied:
As far as I understand, you don't necessarily need Visual Studio, but the Windows Debugging Tools. I've added a link to https://docs.microsoft.com/en-us/windows-hardware/drivers/debugger/ which describes how to install them. I haven't added an explicit dependency to any Chocolatey package providing them, since there are multiple ways to install them and different packages that should do it.

AdmiringWorm replied:
I was under impression that it was part of Visual Studio installation, as mentioned here: https://wiki.qt.io/Qt_Creator_Windows_Debugging .

But the change you made looks good, and I don't really have that much experience with qtcreator to say exactly what is necessary and not.

AdmiringWorm commented on automatic/qtcreatorcdbext/qtcreatorcdbext.nuspec:
```diff
@@ -0,0 +1,34 @@
+<package xmlns="http://schemas.microsoft.com/packaging/2015/06/nuspec.xsd">
+  <metadata>
+    <id>qtcreatorcdbext</id>
+    <version>4.13.2</version>
```
please set the version to `0.0`.
Having it to `4.13.2` will prevent the package to be updated and pushed (since that is the latest version).

michaelweghorn replied:
Done.

AdmiringWorm commented on automatic/qtcreatorcdbext/qtcreatorcdbext.nuspec:
```diff
@@ -0,0 +1,34 @@
+    <id>qtcreatorcdbext</id>
+    <version>4.13.2</version>
+    <packageSourceUrl>https://github.com/admiringworm/chocolatey-packages/tree/master/automatic/qtcreatorcdbext</packageSourceUrl>
+    <owners>AdmiringWorm</owners>
```
Please add a comma `,` and your username (or name if you prefer) to this section.
Since you are the one contributing the initial package.

michaelweghorn replied:
Done.

AdmiringWorm commented on automatic/qtcreatorcdbext/qtcreatorcdbext.nuspec:
```diff
@@ -0,0 +1,34 @@
+    <copyright>© 2020 The Qt Company Ltd.</copyright>
+    <licenseUrl>http://code.qt.io/cgit/qt-creator/qt-creator.git/plain/LICENSE.GPL3-EXCEPT?h=4.1</licenseUrl>
+    <requireLicenseAcceptance>false</requireLicenseAcceptance>
+    <projectSourceUrl>http://code.qt.io/cgit/qt-creator/qt-creator.git/</projectSourceUrl>
```
Please update this link to

```suggestion
    <projectSourceUrl>https://code.qt.io/cgit/qt-creator/qt-creator.git/tree/src/libs/qtcreatorcdbext/</projectSourceUrl>
```

^^ Is the actual source location of the extension.

michaelweghorn replied:
Done.

AdmiringWorm commented on automatic/qtcreatorcdbext/qtcreatorcdbext.nuspec:
```diff
@@ -0,0 +1,34 @@
+    <licenseUrl>http://code.qt.io/cgit/qt-creator/qt-creator.git/plain/LICENSE.GPL3-EXCEPT?h=4.1</licenseUrl>
+    <requireLicenseAcceptance>false</requireLicenseAcceptance>
+    <projectSourceUrl>http://code.qt.io/cgit/qt-creator/qt-creator.git/</projectSourceUrl>
+    <docsUrl>https://doc.qt.io/qtcreator/</docsUrl>
```
This url should be

```suggestion
    <docsUrl>https://wiki.qt.io/Qt_Creator_Windows_Debugging</docsUrl>
```

Which is the only location I can find any documentation of the extension.

michaelweghorn replied:
Done.

AdmiringWorm commented on automatic/qtcreatorcdbext/qtcreatorcdbext.nuspec:
```diff
@@ -0,0 +1,34 @@
+    <docsUrl>https://doc.qt.io/qtcreator/</docsUrl>
+    <mailingListUrl>http://lists.qt-project.org/mailman/listinfo/qt-creator/</mailingListUrl>
+    <bugTrackerUrl>https://bugreports.qt.io/browse/QTCREATORBUG</bugTrackerUrl>
+    <tags>c++ qt ide qtcreator qtcreatorcdbext admin</tags>
```
The tags should include either `extension` or `plugin` (not completely sure which one fits the best).

And the `admin` tag should be removed

michaelweghorn replied:
I've added "plugin" now and removed the "admin" tag.

AdmiringWorm commented on automatic/qtcreatorcdbext/qtcreatorcdbext.nuspec:
```diff
@@ -0,0 +1,34 @@
+    <releaseNotes>[Software Changelog](http://code.qt.io/cgit/qt-creator/qt-creator.git/plain/dist/changes-4.13.2.md)
+[Package Changelog](https://github.com/AdmiringWorm/chocolatey-packages/blob/master/automatic/qtcreatorcdbext/Changelog.md)</releaseNotes>
+    <dependencies>
+      <dependency id="qtcreator" version="4.13.2" />
```
Please change this to:

```suggestion
      <dependency id="qtcreator" version="[4.13.2]" />
```

This will ensure that the cdb extension only accepts the exact version of qtcreator (which would be appropriate since they are updated/provided at the same time).

This is of course something that will need to be also be updated in the `update.ps1` script to replace the version during update.

See https://github.com/AdmiringWorm/chocolatey-packages/blob/develop/automatic/privazer/update.ps1#L8-L10 as an example for that.

michaelweghorn replied:
Done, thanks for the link to the example, that was very helpful!

AdmiringWorm reviewed (changes requested):
In general the changes look good, but there are some things I need you to change before I can merge this package in.

Also, please rename the directory from `qtcreatorcdbext` to `qtcreator-cdbext` (this will be the id of the package).

AdmiringWorm commented:
> Note: I've created this for Qt Creator 4.13.2 already, which is currently only available on the master branch, not develop yet, so this should probably not be integrated into the develop branch before PR #273.

The changes from the master branch have been pulled in to the develop branch now.

AdmiringWorm commented:
@michaelweghorn are you in a position to make the changes I requested?

michaelweghorn commented:
> In general the changes look good, but there are some things I need you to change before I can merge this package in.
> 
> Also, please rename the directory from `qtcreatorcdbext` to `qtcreator-cdbext` (this will be the id of the package).

Thanks for the super-quick review. I have changed the ID and renamed the directory accordingly and tried to address all your comments in the new version.

> The changes from the master branch have been pulled in to the develop branch now.
Thanks, I've rebased this PR on top of current develop branch.

AdmiringWorm commented on automatic/qtcreator-cdbext/qtcreator-cdbext.nuspec:
```diff
@@ -0,0 +1,36 @@
+    <tags>c++ qt ide qtcreator qtcreatorcdbext plugin</tags>
+    <summary>CDB (Microsoft Console Debugger) extension for Qt Creator.</summary>
+    <!-- Do not touch the description here in the nuspec file. Description is imported during update from the Readme.md file -->
+    <description><![CDATA[Qt Creator CDB Extension adds support for the Microsoft Console Debugger (CDB) to Qt Creator, a cross-platform, complete integrated development environment (IDE) for application developers to create applications for multiple desktop and mobile device platforms, such as Android and iOS.
```
Just for future reference, you do not need to update the `<description>` element.
Only the readme.md is necessary to be updated, which will be imported to the description when the package gets updated.

No harm in updating though, just wanted to mention it :smiley:

michaelweghorn replied:
Thanks for the hint. I had actually only manually updated `Readme.md` and the change showed up here as well (without me having thought of it at all) when I tested the update mechanism, so I committed it. :-)

AdmiringWorm reviewed (approved):
Changes LGTM :+1:

Just need to set up a VM to test the package before I can merge this in.
Should be able to do that either tomorrow or during the weekend.

AdmiringWorm commented:
:facepalm: I forgot I had mergify set up.
Oh well.

The package won't be pushed until I merge it into the master branch, which I will do as soon as I have tested the package in the chocolatey VM.

AdmiringWorm commented:
@michaelweghorn your changes have been merged, thanks for your contribution 👍

If you have a chocolatey user account, please let me know which user it is so I can get you added as a co-maintainer of the package when I have pushed the package.

michaelweghorn commented:
> @michaelweghorn your changes have been merged, thanks for your contribution +1

Thanks for your help and for merging!

> If you have a chocolatey user account, please let me know which user it is so I can get you added as a co-maintainer of the package when I have pushed the package.

I have just created an account with the username "michaelweghorn": https://chocolatey.org/profiles/michaelweghorn

AdmiringWorm commented:
Package have been pushed, and I have added you as a co-maintainer.
You should have received an email from chocolatey with a link to confirm being added.

michaelweghorn commented:
Thanks a lot, also for the follow-up improvements you did for the package!

AdmiringWorm commented:
@all-contributors add @michaelweghorn for code
